core/src/main/java/io/substrait/expression/proto/ExpressionProtoConverter.java

```diff
@@ -217,6 +219,28 @@ public Expression visit(io.substrait.expression.Expression.StructLiteral expr) {
         });
   }
 
+  @Override
+  public Expression visit(io.substrait.expression.Expression.UserDefinedLiteral expr) {
+
+    return lit(
+        bldr -> {
+          try {
+            bldr.setNullable(expr.nullable())
+                .setUserDefined(
+                    Expression.Literal.UserDefined.newBuilder()
+                        .setTypeReference(
+                            expr.getType()
+                                .accept(typeProtoConverter)
+                                .getUserDefined()
+                                .getTypeReference())
```

Comment:
I think it would be a bit clearer here if we retrieve the type reference we need directly using
```java
      var typeReference = extensionCollector.getTypeReference(SimpleExtension.TypeAnchor.of(expr.uri(), expr.name()));
```

I had to trace through the type conversion here to verify it did this.

Reply:
done